src/requireJs.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+'use strict';
+
+const requireFromString = require('require-from-string');
+
+function requireJs(content: string, filepath: string) {
+  let result = requireFromString(content, filepath);
+
+  /**
+   * Handle ES Modules
+   */
+  if (typeof result === 'object' && result.__esModule) {
+    if (result.default) {
+      result = result.default;
+    } else {
+      const error = new Error(
+        `${filepath} must use default export with ES Modules`
+      );
+
+      // $FlowFixMe
+      error.code = 'ES_MODULES_DEFAULT_REQUIRED';
```

Comment:
@sudo-suhas I've added an error code here. Flow doesn't like it when you extend ``Error``. Is the ``// $FlowFixMe`` acceptable or should there be a different solution?

Reply:
Setting the code on the instance looks fine to me. I don't think [extending the error class is recommended either](https://github.com/i0natan/nodebestpractices#-22-use-only-the-built-in-error-object). I am not sure if `$FlowFixMe` is required(haven't used flow much myself). @davidtheclark should be able to answer it better.

Reply:
> Can we set and check for a code prop on the error?

@sudo-suhas I added the ``error.code`` here at your request. Did I miss understand what you were asking, or am I incorrectly adding an error code?

Reply:
No, no misunderstanding, you are correctly setting the error code.